lib/puppet/face/config.rb

```diff
@@ -106,6 +106,19 @@
     EOT
 
     when_invoked do |name, value, options|
+      unless name == 'environment' && (options[:section] == 'agent' || options[:section] == 'user')
+        Puppet.warning _(<<-EOM).chomp % {}
+The environment should be set in either the [user] or [agent] section. Variables
+set in the agent section are used when running `puppet agent`. Variables set in
+the user section are used when running various other puppet subcommands, like
+`puppet apply` and `puppet module`. Set the config section by using the
```

Comment:
Maybe add `, and require a local directory environment to exist.`

Reply:
Updated